examples/gateway-hooks/README.md

```diff
@@ -0,0 +1,91 @@
+# Authorization with Grafbase Gateway hooks
+
+This is an example how to implement custom authorization hooks with Grafbase Gateway federation.
+Read more on the hooks from the [gateway hooks documentation](https://grafbase.com/docs/self-hosted-gateway/hooks),
+and on the directive from the [@authorized directive documentation](https://grafbase.com/docs/federation/federation-directives#authorized).
+
+## The components of this example
+
+- `authorized-subgraph` has a simple subgraph server
+- `demo-hooks` contains the code for WebAssembly hooks as a Rust project
+- `federated-schema.graphql` is the federated GraphQL schema
+- `grafbase.toml` has the configuration for the Grafbase Gateway
+
+## Dependencies
+
+To run this example, you need the Grafbase Gateway version 0.4.0 or later, read more how to install it from:
+
+https://grafbase.com/docs/self-hosted-gateway
+
+Additionally, the following tools are needed:
+
+- Rust compiler ([install docs](https://www.rust-lang.org/learn/get-started))
+- Cargo component ([install docs](https://github.com/bytecodealliance/cargo-component?tab=readme-ov-file#installation))
+- A C compiler, such as clang (install based on your system, `cc` command is required)
+- A GraphQL client, such as
+
+For the advanced users using nix with flakes support, all the dependencies excluding the grafbase gateway can be installed with:
```

Comment:
Not in scope for this PR of course, but the gateway should be easy to package.

Reply:
We could have it in the flake.nix of the repo and just include it in the devenv here.